packages/agents/sequencer/src/lib/operations/lighthouse.ts

```diff
@@ -0,0 +1,198 @@
+    });
+  }
+
+  // Ensure that the lighthouse data for this transfer hasn't expired.
+  const status = await cache.lighthousetxs.getLightHouseDataStatus(transferId);
+  if (status !== LightHouseDataStatus.None && status !== LightHouseDataStatus.Cancelled) {
+    throw new LightHouseDataExpired(status, {
+      transferId,
+      lighthouseData,
+    });
+  }
+
+  // Check to see if we have the XCall data saved locally for this.
+  let transfer = await cache.transfers.getTransfer(transferId);
+  if (!transfer || !transfer.origin) {
+    // Get the XCall from the subgraph for this transfer.
+    transfer = await subgraph.getOriginTransferById(origin, transferId);
+    if (!transfer || !transfer.origin) {
+      throw new MissingXCall(origin, transferId, {
+        lighthouseData,
+      });
+    }
+    // Store the transfer locally. We will use this as a reference later when we execute this transfer
+    // in the cycle, for both encoding data and passing relayer fee to the relayer.
+    await cache.transfers.storeTransfers([transfer]);
+  }
+
+  if (!transfer.destination?.reconcile || transfer.destination?.execute) {
+    // This transfer has already been Executed or not Reconciled yet, so slow liquidity is no longer valid.
+    throw new InvalidSlowLiqTransfer({
+      transferId,
+      lighthouseData,
+    });
+  }
+
+  // Lighthouse data needs to be transferred to relayer directly
+  // so we need to estimate tx here to make sure it will not revert
+  try {
+    const destinationDomain = transfer.xparams.destinationDomain;
+    const destinationChainId = await getChainIdFromDomain(destinationDomain, chainData);
+    const relayerAddress = await getGelatoRelayerAddress(destinationChainId);
+    const destinationConnextAddress = config.chains[destinationDomain].deployments.connext;
+
+    const gas = await chainreader.getGasEstimateWithRevertCode(Number(destinationDomain), {
+      chainId: destinationChainId,
+      to: destinationConnextAddress,
+      data: encodedData,
+      from: relayerAddress,
+    });
+
+    logger.info("Estimating a tx success!", requestContext, methodContext, {
+      relayer: relayerAddress,
+      connext: destinationConnextAddress,
+      domain: destinationDomain,
+      gas: gas.toString(),
+      relayerFee,
+      transferId: transferId,
+    });
+  } catch (error: unknown) {
+    throw new GasEstimationFailed({ transferId, lighthouseData });
+  }
+
+  // Create the lighthouse tx in the cache if necessary.
+  await cache.lighthousetxs.setLightHouseDataStatus(transferId, LightHouseDataStatus.Pending);
+  await cache.lighthousetxs.storeLightHouseData(lighthouseData);
+  logger.info("Created a lighthouse tx", requestContext, methodContext, { transferId, lighthouseData });
+
+  const message: Message = {
+    transferId: transfer.transferId,
+    originDomain: transfer.xparams!.originDomain,
+    type: MessageType.SlowPath,
+  };
+
+  await mqClient.publish(config.messageQueue.publisher!, {
```

Comment:
Hmm, is this using the same message queue keys as the fast path? I want to make sure it's fully separated so the type/routingKey should be properly namespaced, so we should make it so it always publishes with a separate name.

Reply:
Yes. What I was thinking here is basically to make the message broker any kind of message, not just execute-fast ones. so we can integrate more like AMB off-chain stuff.